tap_listrak/http.py

```diff
@@ -17,23 +16,36 @@ def get_client(config):
     return client
 
 def log_retry_attempt(details):
-    _, exception, _ = sys.exc_info()
-    LOGGER.info(exception)
-    LOGGER.info('Caught retryable error after %s tries. Message: %s. Waiting %s more seconds then retrying...',
-                details["tries"],
-                exception.message,
-                details["wait"])
+    """Log details about a backoff retry attempt."""
+    exception = details.get("exception")
+    LOGGER.warning(
+        "Retry attempt %s due to error: %s. Waiting %s more seconds before retrying...",
+        details["tries"],
+        str(exception),
+        details["wait"]
+    )
 
+def is_non_retriable_exception(exc):
+    """Avoid retrying on InvalidLogonAttempt errors."""
+    return isinstance(exc, Fault) and "InvalidLogonAttempt" in str(exc)
+
+@backoff.on_exception(
+    backoff.expo,
+    (XMLSyntaxError, TransportError, Fault),
+    max_tries=5,
+    jitter=None,
+    on_backoff=log_retry_attempt,
+    giveup=is_non_retriable_exception # This will prevent retry on bad login
```

Comment:
Since the giveup handler is named `is_non_retriable_exception()`, the comment doesn't add any value and seems redundant. IMO, no comment is needed here.

Reply:
Also please avoid force pushes.